modules/core/src/lib/deck.js

```diff
@@ -403,16 +355,18 @@ export default class Deck {
   // Get the most relevant view state: props.viewState, if supplied, shadows internal viewState
   // TODO: For backwards compatibility ensure numeric width and height is added to the viewState
   _getViewState(props) {
-    return Object.assign({}, props.viewState || this.viewState || {}, {
-      width: this.width,
-      height: this.height
-    });
+    return props.viewState || this.viewState || {};
   }
 
   // Get the view descriptor list
   _getViews(props) {
     // Default to a full screen map view port
-    return props.views || [new MapView({id: 'default-view'})];
+    const views = props.views || [new MapView()];
+    if (views.length && props.controller) {
+      // Backward compatibility: support controller prop
+      views[0].controller = props.controller;
```

Comment:
Issue a `log.deprecated()`?

Reply:
Are we going to deprecate flat props all together? I do like its simplicity when I'm just using the good old single map view.